src/webgpu/api/operation/command_buffer/programmable/programmable_state_test.ts

```diff
@@ -0,0 +1,169 @@
+import { GPUTest } from '../../../../gpu_test.js';
+import { EncoderType } from '../../../../util/command_buffer_maker.js';
+
+interface BindGroupIndices {
+  a: number;
+  b: number;
+  out: number;
+}
+
+type PipelineByEncoderType<T extends EncoderType> = {
+  'non-pass': null;
+  'compute pass': GPUComputePipeline;
+  'render pass': GPURenderPipeline;
+  'render bundle': GPURenderBundleEncoder;
```

Comment:
```suggestion
  'render bundle': GPURenderPipeline;
```
(Though the presence of this error shows that the types here probably aren't helping - maybe usages of PipelineByEncoderType could just be replaced with `GPUComputePipeline | GPURenderPipeline` since the `as` casts are already there. Unless you want to try to make it more typesafe but I'm not sure offhand how to do that.)

Reply:
I'm actually not seeing an error here? (And the CI checks appear to be passing now.)

Reply:
I mean that this looks like a mistake (unless I'm wrong), yet there is no error because this type isn't actually taking effect since it is `as` cast everywhere it's used.

Reply:
Ah, thanks for pointing that out! It was indeed a typo. I reverted to `GPUComputePipeline | GPURenderPipeline` for simplicity.